BookingService/BookingService.Testing/UserServiceTests.cs

```diff
@@ -0,0 +1,315 @@
+        }
+
+        [Theory]
+        [InlineData("validUser", "valid.email@example.com", "Valid1@password", "User")] // Happy path
+        [InlineData("", "valid.email@example.com", "Valid1@password", "User")] // Empty username
+        [InlineData("ab", "valid.email@example.com", "Valid1@password", "User")] // Username too short
+        [InlineData("ThisUsernameIsWayTooLongToBeValidForOurValidationRules", "valid.email@example.com", "Valid1@password", "User")] // Username too long
+        [InlineData("Invalid!User", "valid.email@example.com", "Valid1@password", "User")] // Username with invalid characters
+        [InlineData("validUser", "", "Valid1@password", "User")] // Empty email
+        [InlineData("validUser", "invalid-email", "Valid1@password", "User")] // Invalid email format
+        [InlineData("validUser", "valid.email@example.com", "", "User")] // Empty password
+        [InlineData("validUser", "valid.email@example.com", "short", "User")] // Password too short
+        [InlineData("validUser", "valid.email@example.com", "NoUpperCase1@", "User")] // Password without uppercase
+        [InlineData("validUser", "valid.email@example.com", "NOLOWERCASE1@", "User")] // Password without lowercase
+        [InlineData("validUser", "valid.email@example.com", "NoNumber@", "User")] // Password without number
+        [InlineData("validUser", "valid.email@example.com", "NoSpecialCharacter1", "User")] // Password without special character
+        [InlineData("validUser", "valid.email@example.com", "Password with space1@", "User")] // Password with spaces
+        [InlineData("validUser", "valid.email@example.com", "Valid1@password", "")] // Empty role
+        [InlineData("validUser", "valid.email@example.com", "Valid1@password", "InvalidRole")] // Invalid role
+        [InlineData("", "", "", "")] // All fields empty
+        [InlineData("validUser", "valid.email@example.com", "Valid1@password", "Admin")] // Valid with Admin role
+        [InlineData("validUser", "valid.email@example.com", "Valid1@password", "Host")] // Valid with Host role
+        public async Task RegisterAsync_ShouldNotProceed_WhenDataInvalid(string username, string email, string password, string role)
+        {
+            //Arrange
+            var registerUserDto = new RegisterDto { Username = username, Email = email, Password = password, Role = role };
+            var user = new User();
+            var validationResult = _registerDtoValidator.Validate(registerUserDto);
+
+            if (!validationResult.IsValid)
+            {
+                _mockCustomUserManager.Verify(userManager => userManager.CreateAsync(user, registerUserDto.Password), Times.Never);
+                _mockCustomRoleManager.Verify(roleManager => roleManager.RoleExistsAsync(registerUserDto.Role), Times.Never);
+                _mockCustomUserManager.Verify(userManager => userManager.AddToRoleAsync(user, registerUserDto.Role), Times.Never);
+
+                return;
+            }

```

Comment:
It obviously will be called never, because u apply these assertions BEFORE calling `RegisterAsync`.
The same: remove if condition and move validation assertions to other classes